mithril-signer/src/runtime/state_machine.rs

```diff
@@ -23,21 +23,19 @@ pub enum SignerState {
         epoch: Epoch,
     },
 
-    /// `Registered` state. The Signer has successfully registered against the
-    /// Aggregator for this Epoch, it is now able to sign.
-    Registered {
-        /// Epoch when Signer may sign.
-        epoch: Epoch,
+    /// `ReadyToSign` state. The signer is registered and ready to sign new messages.
+    ReadyToSign {
+        /// Time point when signer transited to the state.
+        time_point: TimePoint,
+
+        /// Last signed entity type that the signer signed.
```

Comment:
It's the last signed entity type since it was in ReadyToSign.
When the state machine exit the ReadyToSign state, the last_signed_entity_type is cleared

Reply:
Done ✅